application/client.plugins/projects/serial/src/lib/views/dialog/port.available/components.ts

```diff
@@ -0,0 +1,197 @@
+// tslint:disable:no-inferrable-types
+
+import { Component, OnDestroy, ChangeDetectorRef, Input, AfterViewInit } from '@angular/core';
+import { IPortInfo, IPortState } from '../../../common/interface.portinfo';
+import { IOptions } from '../../../common/interface.options';
+import { Logger } from 'chipmunk.client.toolkit';
+import Chart from 'chart.js';
+import { Observable, Subscription } from 'rxjs';
+import Service from '../../../services/service';
+
+interface IConnected {
+    port: IPortInfo;
+    options: IOptions;
+    state: IPortState;
+}
+
+interface Irgb {
+    red: number;
+    green: number;
+    blue: number;
+    opacity: 1;
+}
+
+@Component({
+    selector: 'lib-dia-port-available-com',
+    templateUrl: './template.html',
+    styleUrls: ['./styles.less']
+})
+
+export class DialogAvailablePortComponent implements OnDestroy, AfterViewInit {
+
+    @Input() public port: IPortInfo;
+    @Input() public observer: { event: Observable<any> };
+    @Input() public connected: IConnected[];
+    @Input() public spyState: { [key: string]: number };
+
+    private _subscriptions: { [key: string]: Subscription } = {};
+    private _canvas: HTMLCanvasElement;
+    private _ctx: any;
+    private _step = 10;
+    private _animation = 5000;
+    private _read: number;
+    private _chart: Chart;
+    private _spark = Array<number>(this._step + 1).fill(0);
+    private _logger: Logger = new Logger(`Plugin: serial: inj_output_bot:`);
+    private _destroyed: boolean = false;
+
+    constructor(private _cdRef: ChangeDetectorRef) {
+    }
+
+    ngAfterViewInit() {
+        if (!this._ng_isConnected(this.port)) {
+            this._read = 0;
+            this._canvas = document.getElementById(`canvas_${this.port.path}`) as HTMLCanvasElement;
+            this._createChart().then(() => {
```

Comment:
Please use @ViewChild and ElementRef from Angular to get access to node (canvas in your case). Try to avoid from direct calling DOM while we are within Angular